File: src/main/java/com/example/mediaApp/model/entity/Post.java
package com.example.mediaApp.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.List;

@Entity(name = "Post")
@Table(name = "post")
@Setter
@Getter
@NoArgsConstructor
@ToString(onlyExplicitlyIncluded = true, includeFieldNames = false)
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private AppUser user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id")
    private Group group;

    private String text;

    @Lob
    @Column(length = 1000000)
    private byte[] image;

    // Unidirectional
    @OneToMany(orphanRemoval = true)
    @JoinColumn(name = "post_id")
    private List<Like> likes;

    // Unidirectional
    @OneToMany(orphanRemoval = true)
    @JoinColumn(name = "post_id")
    private List<Comment> comments;

    @OneToMany(
            mappedBy = "id",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private List<Report> reports;
}

File: src/main/angular-media-app/src/app/models/post.ts
import { User } from "./user";
import { Group } from "./group";
import { Like } from "./like";
import { Comment } from "./comment";

export class Post {
    id: number;
    user: User | null;
    group: Group | null;
    text: string;
    image: number[] | null;
    likes: Like[];
    comments: Comment[];
    imgSrc: string;
    isCommenting: boolean;

    constructor(){
        this.id = 0;
        this.user = new User();
        this.group = new Group();
        this.text = "";
        this.image = null;
        this.likes = [];
        this.comments = [];
        this.imgSrc = '';
        this.isCommenting = false;
    }

    static convertNewPost(post: Post): Post{
        let newPost: Post = new Post();
        newPost.id = post.id;
        newPost.user = post.user;
        newPost.group = post.group;
        newPost.text = post.text;
        newPost.image = post.image;
        newPost.likes = post.likes;
        newPost.comments = post.comments;
        newPost.imgSrc = newPost.getImageSrc();
        newPost.isCommenting = false;
        return newPost;
    }

    getImageSrc(): string{
        if(this.image){
            return 'data:image/jpeg;base64,' + this.image;
        }
        return '';
    }
}
